Add Modrinth modpack support

Pyroglyph opened the pull request:
## Purpose
Currently, GDLauncher does not support downloading mods/modpacks hosted on the [Modrinth](https://modrinth.com/) platform. This feature has been requested in issue #768.

I'm aware of GDevs' other project [ferium](https://github.com/gorilla-devs/ferium) which already supports Modrinth, and also the [`gdlib-refactor` branch](https://github.com/gorilla-devs/GDLauncher/tree/gdlib-refactor) (which I assume may eventually succeed GDLauncher?), but that branch is far from completion.

## Approach
This PR allows users to search for and install mods and modpacks hosted on Modrinth in the same way that they can currently be installed from CurseForge and FTB.

![image](https://user-images.githubusercontent.com/9498104/171962285-2d8df8f9-f2e0-4156-8c92-0b0b094f861c.png)

The implementation is highly based on the existing CurseForge/FTB installer code so it adds no new dependencies, which I see as an advantage.

#### Open Questions and Pre-Merge TODOs
- [x] Add support for searching for and installing modpacks from Modrinth
- [x] Add support for searching for and installing individual mods from Modrinth
- [x] Add a proper way to show errors to the user (i.e. when a mod fails to download, or when the user tried to install an unsupported pack type like Quilt)
- [ ] [Modrinth provides fallback download links for overrides.](https://docs.modrinth.com/docs/modpacks/format_definition/#downloads) If an override file fails to download, the launcher should retry the download using given alternate links.
- [x] Should we send a Google Analytics event for Modrinth API calls?

## Learning
I have read the Modrinth API docs (linked below).

#### Resources
- [Modrinth API Spec](https://docs.modrinth.com/api-spec/)
- [Modrinth Modpack Format](https://docs.modrinth.com/docs/modpacks/format_definition/)

blarfoon committed:
Fix changelogs discord button

baibaichris committed:
improve add account modal (#917)

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
fix menu overflow

blarfoon committed:
Update all deps + native modules to NAPI

blarfoon committed:
fix linting

blarfoon committed:
downgrade node

blarfoon committed:
fix upload

blarfoon committed:
fix macos 7z path

blarfoon committed:
v1.1.9

blarfoon committed:
v1.1.10

Ladvace committed:
mcVersionChangerFix (#946)

Ladvace committed:
fix (#954)

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

Ladvace committed:
empty mods list message fix (#955)

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

meponderR committed:
Increase Maximum Value on Memory Sliders  (#950)

* Change Maximum Value for Max Memory in Global Settings

* Change Maximum Value for Max Memory in Instance Settings

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

tilda committed:
fix minor spelling mistake in mod sync (#935)

s/syncronizing/synchronizing

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

kevinoconnor7 committed:
Parse fabric loader version from modloader ID (#949)

CurseForge doesn't (or no longer) users the yarn property in the
manifest to specify the fabric version. Instead they now include the
version in id property, ex: `fabric-0.11.6`.

When importing an instance we'll check for the yarn property first, for
backwards compatibility, but will then try to extract it from the id if
it's missing.

For exporting we'll now include the loader version in the id and omit
the yarn property, exactly like CurseForge.

Fixes #942

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
1.1.11

Eskaan committed:
[Inconsitency] Concurrent downloads setting not used whit ftb modpacks | Better progress update (#962)

* Changed updatePercentage() function to defaut

Changed the function to what all the other Downloads use.
- Checking if progress changed => less CPU load
- Using parseInt() function to prevent errors

* Applied concurrent download setting to FTB file-download on instance creation

* Codacity means I should do that :/

Eskaan committed:
Added usual MacOS default menu (#944)

* Added usual MacOS default menu

New Features:
 - option to open About Page
 - Services Submenu
 - Hide Others
 - Show All

* Changed requested Edits

- Wrong quote on line 78
- Revert to original click lambda

* …And another wrong quote

* ...fixed wrong quote

Wow, seems that this happend quite often :/

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

Eskaan committed:
Corrected Lint error - Original Commit by @MrHallway0 (#964)

Co-Authored-By: Mr. Hallway <git@mrhallway.me>

Co-authored-by: Mr. Hallway <git@mrhallway.me>

rastiqdev committed:
Update Screenshots.js (#966)

Eskaan committed:
Fixed Problem whit hidden accounts in modal (#959)

* Fixed Problem whit hidden accounts in modal

Just had to add overflow: auto; and a padding to the parent element.

* Moved CSS-Fix for Account Modal to container initialization

Co-Authored-By: Mr. Hallway <git@mrhallway.me>

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>
Co-authored-by: Mr. Hallway <git@mrhallway.me>

Eskaan committed:
Limited memory sliders to max system memory (#967)

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

Eskaan committed:
Now accurate Kb to Mb calculation (#1003)

kevinoconnor7 committed:
Auto-detect CurseForge modpack loader type (#980)

* Auto-detect CurseForge modpack loader type

Currently all CurseForge modpacks are assumed to be using Forge, whereas
they may actually be using Fabric. Instead of hardcoding which one is
being used, we can parse the type from the manifest.

Fixes #961

* Support changes versions of fabric modpacks

A few refactors were needed to enable this:

  - Created a utility for extracting the fabric version from a manifest
  - Revised the change version flow to properly parse fabric vs forge
    loader versions (previously it just assumed it was a forge modpack)
  - Reworked the CreateInstance flow to better support code-sharing

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
Update FUNDING.yml

blarfoon committed:
Update README.md

blarfoon committed:
Update README.md

blarfoon committed:
Add social links to the setting's sidebar (#1012)

* Add social links

* Fix eslint

blarfoon committed:
Add security check for accounts

blarfoon committed:
drastically improve addinstance modal performance

blarfoon committed:
Remove unused import

Eskaan committed:
Style fix (#1014)

Scaled ko-fi.png to patreon.png

blarfoon committed:
Update dependencies

blarfoon committed:
Fix text flicker

blarfoon committed:
lazy load instance manager & settings

blarfoon committed:
downgrade electron-builder

blarfoon committed:
potential build fix

blarfoon committed:
Revert "Update dependencies"

This reverts commit 326923d754ce6bfee3855cc23f6f0cdf58f1c95f.

blarfoon committed:
Fix package lock

blarfoon committed:
update deps & fix ant design clown circus

blarfoon committed:
Fix electron-builder

blarfoon committed:
fix lint

blarfoon committed:
Update changelog modal design

blarfoon committed:
Fix Ganalytics

blarfoon committed:
Remove unused imports

blarfoon committed:
Add entry in changelog

blarfoon committed:
Possibly fix CI

blarfoon committed:
Fix CI (?)

blarfoon committed:
test

blarfoon committed:
fix ci

blarfoon committed:
test

blarfoon committed:
disable lint

blarfoon committed:
Fix CI

blarfoon committed:
more testing

blarfoon committed:
more test

blarfoon committed:
fix CI

Ladvace committed:
curseForge api fix (#1022)

* fix courseForge api

* don't include modLoaderType if not fabric

blarfoon committed:
fix CI

blarfoon committed:
Update licensing

blarfoon committed:
Update settings sidebar

hypherionmc committed:
Added Instance Duplication Feature (#1002)

* Added Instance Duplication

* Removed Sync functions and removed ESLINT disable line

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
Fix duplicate instance

blarfoon committed:
hotfix for duplicate instance

blarfoon committed:
fk CRA

Ladvace committed:
Java 16 support (#993)

* init

* version comparison, steps in the automatic java setup

* javaPath selectors merged, isLatestJavaDownloaded functions merged

* all java version in one folder

* g

* Non-Windows unpack fix
variable not renamed and tried to unpack three times

* New overview progress bar

* Auto stash before merge of "java16" and "origin/java16"

* Simple fixes

* test fix

* Auto stash before merge of "java16" and "origin/java16"

* test fix

* do not downloaded if already installed

* Fixed last commit - now working correct

* Squashed java installation into function to simplify the code

* Added button to force full reinstall
*only outdated Java will be updated whiteout force install

* Fixed hole java directory deletion to only java<version> directory.

* check if java already exist + refactor + force installation

* t

* Corrected variable name java8 to java16

* Removed option to force the installation

* Removed option to force the installation

* Removed option to force the installation

* javaSetup always showing fix

* logic fix

* Grammar and Inconsistencies

* Windows Overview bar fix

* Set default java path version
Fixed forge installer - now working again

* Forge installer now giving version to prevent future bugs

* fix

* progressbar color

* sidebar stylefix

* showing missing java versions

* Missing versions style

* style

* fix progress bar

* fix

* removed usless latest java version check

* f

* Fix download steps...

* Fix taskbar progress...

* Fixed text and memoized javasetup component

* progress bar color

* f

* f

* java automatic detection and java paths in redux fix

* show if already installed

* java log details if already installed (in the automatic setup)

* custom java path based on mcversion

* prevent java setup modal from closing if opened by user + linting fix

* f

* close button only when already installedcand automatic close at the end of installation

* close button fix + show java version in the custom java path

* Finalize java

* Remove unused imports

Co-authored-by: Code-Ac <76821666+Code-Ac@users.noreply.github.com>
Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>
Co-authored-by: Davide <davide.ceschia@gmail.com>

blarfoon committed:
Hotfix startup java check

blarfoon committed:
reenable lint

hypherionmc committed:
Fix for Beta Channel not working (#1025)

* Fix formatting that eslint is yelling at me about

* Added Fix for Beta Channel not Working

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
privacy policy & terms and conditions

blarfoon committed:
acceptable use policy

blarfoon committed:
Add policies to login page

blarfoon committed:
Add support for forge 1.17

blarfoon committed:
nitpick

blarfoon committed:
Fixed custom java issue

blarfoon committed:
another java fix

Eskaan committed:
lint fix (#1030)

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

Ladvace committed:
login overlapping text fix (#1032)

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

Ladvace committed:
discord RPC - current playing instance (#1033)

* basic discord rpc running instance

* removed useless code

* fix

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
Fix discord RPC

blarfoon committed:
v1.1.12

blarfoon committed:
v1.1.13

blarfoon committed:
Simplify modal animation

blarfoon committed:
Fix crash

blarfoon committed:
Hotfix export issue

blarfoon committed:
Fix crash on instance rename

blarfoon committed:
Fix html entities in news not being decoded

blarfoon committed:
Fix eslint

blarfoon committed:
v1.1.14

blarfoon committed:
Fix node-7z

blarfoon committed:
Fix eslint

blarfoon committed:
Fix eslint again

blarfoon committed:
track cf/ftb api usage

Eskaan committed:
Codeac hotfix (#1045)

* npm install

* ignore MacOS system folder ".DS_Store"

* modIcon fully shown

* error detected by IDE

* fixed error when beta channel is selected

* lint fix

blarfoon committed:
v1.1.15-beta.1

Eskaan committed:
Logo style fix for #605 (#1047)

blarfoon committed:
Fix portable updates

blarfoon committed:
possibly fix fingerprint match being null

blarfoon committed:
v1.1.15-beta.2

blarfoon committed:
Fix startup mods check

blarfoon committed:
Simplify 7z process

blarfoon committed:
start rollout custom apis

blarfoon committed:
fix eslint

blarfoon committed:
temporarily disable custom apis

blarfoon committed:
Removed patreon links from everywhere

blarfoon committed:
removed patreon link from changelog

blarfoon committed:
add button to check for mod updates

blarfoon committed:
don't track changelog in beta

blarfoon committed:
v1.1.15-beta.3

huantianad committed:
Fix Icons in .deb Release (#1039)

* Try to fix .deb icon location

* Actually fix deb icons this time

* Crap I should've uncommented that

* Ok finally done it, moved all icons to seperate folder

* replaced blurry icons with non-blurry ones

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

furtabs committed:
Mods Browser: Text to Font Awesome Icons for Project Buttons (#1035)

* Mods Browser: Text to Font Awesome Icons for Project Buttons

* ESLint is now passing

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

hypherionmc committed:
Happy now Codacy? (#1038)

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
electron 14

blarfoon committed:
updated gdlib

blarfoon committed:
Added new mc startup method

Eskaan committed:
"DAMn" - Davide 2021 (#1061)

blarfoon committed:
Batch multiple CF requests together

blarfoon committed:
Fix startup crash for beta

Eskaan committed:
Settings refactor (#1063)

* Style inconsistencies & typo

* Socials Styling (now squares)

* Settings Refactor (better look)

* Lint fix (caused by master)

blarfoon committed:
Fix for macos

blarfoon committed:
Fix eslint

blarfoon committed:
fix eslint pt. 2

blarfoon committed:
Remove else...

Ladvace committed:
image "squashed" fix + crash fix while switching tab fast (#1074)

Eskaan committed:
README.md better badge overview & line break at LICENCE to make it identifiable for shields.io (#1064)

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
v1.1.15-beta.4

oitsjustjose committed:
Add Instance Sorting Options (#887)

* Init changes to sortability

* Fix missing reducer (it was there before???)

* 0-origin indexing

* Fix uneccessary console log

* Remove change to eslint-ignored line

* Fix bad merge

* Fix linting error

* Move useSelector

* Memoize instances with sort to improve performance

* Reduce diff in PR

blarfoon committed:
Fix discord link

blarfoon committed:
Fix readme

blarfoon committed:
Improve readme performance

coehlrich committed:
Update electron to 14.1.0 (#1087)

Fixes forge installer not downloading

huantianad committed:
Fix wording and punctuation of settings descriptions. (#1080)

* Fix wording and puncutation of settings descriptions.

* Thank you eslint

* More slight wording changes

blarfoon committed:
v1.1.15-beta.5

huantianad committed:
Initial implementation of open export folder (#1082)

* Initial implementation of open export folder

* move open function inside thirdstep

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
Electron 15 + macos 7z path fix

blarfoon committed:
v1.1.15-beta.6

Eskaan committed:
Remove Symlink option for non-Windows (#1097)

* Memory Slider style fix

* Don't make the Symlink option accessible to non-Windows
Doesn't work on MacOS / Linux
Is not needed anyways there

* Fix Changelog same key error

Eskaan committed:
v1.1.15-beta.7 (#1128)

* Put changelog notes into gh release body

* v1.1.15-beta.7
Test version for gh releases

* Fix a few possible bugs

* Hope ESLint is still okay

* ESLint?

* Pls work

* Please ESLint I am doing this on my mobile

* Increase code legibility

* Further code improvements

Co-authored-by: Davide <davide.ceschia@gmail.com>

Eskaan committed:
v1.1.15 (#1127)

* Added extended information (commit/pr/comment) to the changelog

* Changelog

* Remove unused divs

* Regards, your Developers

* Re-Added v1.1.15 changelog

* Revert "Regards, your Developers"

This reverts commit dc46da74bbf07281f8d47516cbbb4901e1f224af.

blarfoon committed:
v1.1.15

blarfoon committed:
Fix undefined PR

blarfoon committed:
Migrate to java 17 (latest java) + sponsors

blarfoon committed:
Fixed microsoft login when offline

blarfoon committed:
Better design for startup modal

blarfoon committed:
v1.1.16-beta.1

blarfoon committed:
Update changelog

blarfoon committed:
Fix extract

blarfoon committed:
test ga_id

blarfoon committed:
test ga_id 2

blarfoon committed:
test ga_id 3

blarfoon committed:
test ga_id 4

blarfoon committed:
Defer analytics loading

blarfoon committed:
pass env to main thread

blarfoon committed:
sentry implementation

blarfoon committed:
v1.1.16-beta.2

blarfoon committed:
forgot secret

blarfoon committed:
FIx sentry dist

blarfoon committed:
undo analytics defer

blarfoon committed:
Prevent discord reset when client is not set

blarfoon committed:
include all source maps

blarfoon committed:
Improve create server ad

blarfoon committed:
Fix java modal always appearing

blarfoon committed:
Reverted play button

blarfoon committed:
Fixed ad modal mixing with crash modal

blarfoon committed:
v1.1.16-beta.4

blarfoon committed:
Temporary potential fix for log4j vulnerability

blarfoon committed:
v1.1.16

blarfoon committed:
Typo..

blarfoon committed:
Removed timed close ad

blarfoon committed:
Fixed "update all mods" button

blarfoon committed:
v1.1.17

blarfoon committed:
Properly fix log4j 0 day bug

blarfoon committed:
v1.1.18

blarfoon committed:
FIx artifacts upload to sentry

blarfoon committed:
Definitive log4j fix

blarfoon committed:
Check resources on instance launch

blarfoon committed:
Update dependencies and deprecate old cdn

blarfoon committed:
Fix package lock

blarfoon committed:
Prevent infinite loop

blarfoon committed:
Fix edge cases

blarfoon committed:
v1.1.19

blarfoon committed:
Merry Christmas to y'all from the GDL Team!

blarfoon committed:
Rollback antd version due to regression in performance

blarfoon committed:
better analytics

blarfoon committed:
Add repair functionality

blarfoon committed:
Update changelog

blarfoon committed:
Fix instance repair

blarfoon committed:
v1.1.20

Eskaan committed:
Fixed macOS and Linux update check

Eskaan committed:
Merge pull request #1222 from Code-Ac/codeac-update-fix

Fixed Macos and Linux updates (#1222)

EliTheCoder committed:
fix incorrect benchmark times (#1233)

LilyRose2798 committed:
Add better ram slider (#1227)

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
v1.1.21

blarfoon committed:
decrease traces sample rate

blarfoon committed:
initial work for API migration

blarfoon committed:
Fix curseforge zip import

blarfoon committed:
fix matched fingerprints

blarfoon committed:
fix package fingerprint name

blarfoon committed:
Fix MC startup for old versions

blarfoon committed:
Improve java memory management system

blarfoon committed:
Add play button to instance manager

blarfoon committed:
Add categories to mods

blarfoon committed:
Update a few dependencies

blarfoon committed:
Change tabs style

blarfoon committed:
switch to preact

blarfoon committed:
disable sentry + fix mc version in new apis

blarfoon committed:
v1.1.22-beta.1

blarfoon committed:
restore CF tracking

blarfoon committed:
Updated changelogs

blarfoon committed:
announcement in home

blarfoon committed:
do not upload sourcemaps

blarfoon committed:
Change announcement color

Ladvace committed:
new curseForge enpoint + secret key from gitHub action (#1334)

* new curseForge enpoint + secret key from gitHub action

* mispelling fix, forgive me pls

* fix

* fix

* test

* test

Co-authored-by: Gianmarco Cavallo <gianmarco.cavallo.consulente@bccsi.bcc.it>

blarfoon committed:
Disable digital signature

blarfoon committed:
v1.1.22-beta.2

blarfoon committed:
v1.1.23

Pyroglyph committed:
Add Modrinth modpack support

Pyroglyph committed:
Fix some modpacks being unable to be downloaded

Pyroglyph committed:
Add support for installing individual mods from Modrinth

I've also added some JSDoc types so I don't go insane

Pyroglyph committed:
Refactor Modrinth API, better naming

Pyroglyph committed:
JSDoc typedefs are now available project-wide!

Ladvace committed:
Workaround for opted out CF mods (#1345)

* fix

* fix

* optedOutModsList modal

* mod loading optedout modal

* wip

* wip

* close secondary window crash fix

* console.logs removed

* fix crashes

* Fix duplicated mods when repairing

* wip

* disabled ftb

* removed console log

* download specific version mods and resourcepacks

* fix resourcepack folder and download timeout

* hide modpack tab if the instance is ftb

* modManifest.id instead of addon.mainFileId

* cancel button to the left

* skip 404 mods + name fix

* opted out single mods fix

* remove mod from config when single mod download fail

* mod installing fix

* update optedout mod fix

* switch version mod fix

* disable switch button if page not loaded

* electron update

* addon download fix + mini refactor

* remove mod restore point

* fix user agent

* fix styling issue

* fix: encode url before download

* better safety checks

* fix: user agent crash fix

Co-authored-by: Gianmarco Cavallo <gianmarco.cavallo.consulente@bccsi.bcc.it>
Co-authored-by: Davide <davide.ceschia@gmail.com>

blarfoon committed:
v1.1.24

blarfoon committed:
Increase mod download timeout

blarfoon committed:
Fix encoding urls

blarfoon committed:
increase mod download timeout

blarfoon committed:
Cloudflare fix and initial support for mac arm64

blarfoon committed:
v1.1.25

Eskaan committed:
Fixed Changelog Generation. Nothing could go wrong (#1364)

Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

blarfoon committed:
clear cookies on each run

Ecorous reviewed (approved):
Looks good!

Ladvace commented on src/common/modals/AddInstance/InstanceName.js:
```diff
@@ -288,6 +308,67 @@ const InstanceName = ({
           ramAmount ? { javaMemory: ramAmount } : null
         )
       );
+    } else if (isModrinthModpack) {
```
Why not just doing `if(modpack?.project_type)` or just `const isModrinthModpack = modpack?.project_type;`

Pyroglyph replied:
I didn't write `if (modpack?.project_type)` because it's less readable. Someone who doesn't know anything about the code will look at that and think "why am I doing this stuff if `project_type` is truthy?".

I also didn't do `const isModrinthModpack = modpack?.project_type;` because the name `isModrinthModpack` implies that it's a boolean, but it could also be a string (because `project_type` is a string). This removes that potential confusion.

I'm a big fan of so-called "self-documenting code", so I try and write in such a way that makes it as easy as possible for other people to read and understand.

Ladvace committed:
normalizeModData when downloading optedout mods (#1376)

Co-authored-by: Gianmarco Cavallo <gianmarco.cavallo.consulente@bccsi.bcc.it>

Ladvace committed:
antd upgrade (#1371)

Co-authored-by: Gianmarco Cavallo <gianmarco.cavallo.consulente@bccsi.bcc.it>
Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

Ladvace committed:
Open mod page button doesn't work #1373 (#1375)

Co-authored-by: Gianmarco Cavallo <gianmarco.cavallo.consulente@bccsi.bcc.it>
Co-authored-by: Davide Ceschia <10165264+killpowa@users.noreply.github.com>

Pyroglyph committed:
Fix single modpacks returning from the Modrinth API as arrays

Pyroglyph commented on src/common/modals/AddInstance/InstanceName.js:
```diff
@@ -103,6 +122,7 @@ const InstanceName = ({
+    const isModrinthModpack = Boolean(modpack?.project_type);
     let manifest;
 
     // If it's a curseforge modpack grab the manfiest and detect the loader
```
I did not write that comment, it's not part of this PR.

Pyroglyph committed:
Turn on GA for Modrinth

Pyroglyph committed:
Fix Modrinth mods not showing as installed + fixed mod overview

Pyroglyph committed:
Unrecoverable errors now show a message to the user (instead of console)

Ladvace commented on src/common/modals/AddInstance/InstanceName.js:
```diff
@@ -288,6 +310,112 @@ const InstanceName = ({
+      }
+
+      dispatch(
+        addToQueue(
```
in this PR: https://github.com/gorilla-devs/GDLauncher/pull/1345 I modified `downloadFunction `and `addToQueue`, try giving a look if nothing breaks (with the curseforge logic)

Ladvace replied:
you also need to solve conflictscts with the master branch

Pyroglyph replied:
Resolved the merge conflicts and downloading Modrinth packs still seem to work fine with your changes!

Pyroglyph committed:
Merge branch 'master' into modrinth-support

blarfoon committed:
feat: new napi implementation

blarfoon committed:
:(

blarfoon committed:
fix :(

blarfoon committed:
mock napi files

blarfoon committed:
Update readme

blarfoon committed:
Merge branch 'master' into modrinth-support

Pyroglyph committed:
Add heuristic scanning for mrpack files

This can be removed once Modrinth begin blocking potentially malicious scripts/binaries

Pyroglyph committed:
Revert "Add heuristic scanning for mrpack files"

This reverts commit ab0452434269293e50ae47110b98d838b48b0785.

Pyroglyph committed:
Re-add imports that I removed by accident

Whoops

Pyroglyph committed:
Properly use the mrpack manifest to download mods

Pyroglyph committed:
Fix MR pack updates not downloading new mods

Pyroglyph committed:
Appease ESLint
